packages/edit-site/src/components/layout/router.js

```diff
@@ -17,38 +17,82 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
 import SidebarNavigationScreenGlobalStyles from '../sidebar-navigation-screen-global-styles';
 import SidebarNavigationScreenMain from '../sidebar-navigation-screen-main';
 import SidebarNavigationScreenNavigationMenus from '../sidebar-navigation-screen-navigation-menus';
-import SidebarNavigationScreenPage from '../sidebar-navigation-screen-page';
 import SidebarNavigationScreenTemplatesBrowse from '../sidebar-navigation-screen-templates-browse';
-import SidebarNavigationScreenTemplate from '../sidebar-navigation-screen-template';
-import SidebarNavigationScreenPattern from '../sidebar-navigation-screen-pattern';
 import SidebarNavigationScreenPatterns from '../sidebar-navigation-screen-patterns';
 import SidebarNavigationScreenNavigationMenu from '../sidebar-navigation-screen-navigation-menu';
 import DataViewsSidebarContent from '../sidebar-dataviews';
 
 const { useLocation, useHistory } = unlock( routerPrivateApis );
 
-export default function useLayoutAreas() {
-	const isSiteEditorLoading = useIsSiteEditorLoading();
+function useRedirectOldPaths() {
```

Comment:
This hook centralizes the redirection of the old paths to the new paths. 

@jsnajdr I noticed a bug though, if I type a new url while the site editor is loaded, the sidebar of the site editor doesn't update properly but if I refresh the page, it does render the right sidebar. (So the redirect was successful but something is not re-rendering properly as the redirect happens)

Reply:
There is also the case with `http://localhost:8888/wp-admin/site-editor.php?postId=twentytwentyfour%2F%2Fwp-custom-template-sss&postType=wp_template`. Previously we would render the page with the details panel, and now it goes to the list. The params are effective visually in `pages` where we have the `list` view as default, but not in templates etc.. Should we redirect to `edit` mode in these cases or remove the params altogether? 🤔

Reply:
We have a couple options: we can switch the templates to list view as well or redirect to "edit". Leaning towards edit for now.

Reply:
On a second though, I'm not sure we should redirect here to another page here. This URL is valid even in the new url scheme. What we're missing is solving the bug of the "controlled selection". In other words, even in the table block, the "row" should appear as selected.

Reply:
> There is also the case with `postId=...&postType=wp_template`

This URL, if it doesn't have `canvas=edit`, now means "show the list of posts of type `wp_template`, and select the `postId` there if you can." For `postType=page` it actually highlights something, but for other types the list UI doesn't use the `postId` parameter. But it could.

> if I type a new url while the site editor is loaded, the sidebar of the site editor doesn't update properly

@youknowriad This is weird, can you post a video, or specify which URL you were on and which new URL you typed?

Reply:
@jsnajdr Actually, it's broken for me even when loading a page in a new tab:

```
http://localhost:8888/wp-admin/site-editor.php?path=%2Fwp_template
```

Reply:
Ok I found the issue, it's a race condition between the effect in The RoutesProvider and the effect here. child effects run before parents. I'll see if I can come up with the correct fix.

Reply:
Solved here [2af8009](https://github.com/WordPress/gutenberg/pull/61741/commits/2af80090922e45bafb9026ada695bd1a751eb27c)